### PR TITLE
added new json functions

### PR DESCRIPTION
**Pull request by awakchau-tibco**
Added new json functions like get, length, numbersToString, objKeys, objValues and set.

### CODE REVIEW

File: function/json/descriptor.json [hunk 1]
```diff
@@ -1,11 +1,87 @@
...
+          "type": "string"
+        }
+      ],
+      "return": {
+        "type": "any"
+      }
+    },
+    {
+      "name": "length",
+      "description": "Get the number of elements in a json object",
```
Comment (lixingwang) [inline function/json/descriptor.json:29]: length should also compatible to array

File: function/url/encode.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	if len(u.Host) > 0 {
+		encoded = append(encoded, u.Host)
+	}
+	baseURL, err := url.Parse(strings.Join(encoded, ""))
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse base url [%s]: %s", baseURL, err.Error())
+	}
+	baseURL.Path += u.Path
+	if len(u.Fragment) > 0 {
+		baseURL.Fragment += u.Fragment
```
Comment (lixingwang) [inline function/url/encode.go:56]: we should also escape fragment, such as path also need to be escaped

**Review by lixingwang — CHANGES_REQUESTED** [review]

File: function/json/descriptor.json (hunk 1 above)
Comment (awakchau-tibco) [reply function/json/descriptor.json]: updated the json function description to support array

File: function/url/encode.go (hunk 2 above)
Comment (awakchau-tibco) [reply function/url/encode.go:56]: @lixingwang we already escape path, query and fragment part of the URL. I've updated the test case to include all 3 parameters. For example:
`Input : https://subdomain.example.com/path to be escaped?q=hello world#fragment with space`
`Output: https://subdomain.example.com/path%20to%20be%20escaped?q=hello+world#fragment%20with%20space`

**Review by lixingwang — APPROVED** [review]

**Pull request merged**